File: script/fake-ghc.py
#!/usr/bin/python

import sys
import os

if __name__ == "__main__":
  argv = sys.argv
  argv.pop(0)
  if "--numeric-version" in argv:
    os.system("ghc --numeric-version")
    sys.exit()
  print "== GHC Arguments: Start =="
  for arg in argv:
    print arg
  print "== GHC Arguments: End =="

File: script/fake-ghc.hs
{-#LANGUAGE CPP #-}

import System.Environment
import System.Info
import Data.Version
import Control.Monad

main = do args <- getArgs
          if "--numeric-version" `elem` args
            then putStrLn $ showVersion compilerVersion
            else do putStrLn "== GHC Arguments: Start =="
                    mapM putStrLn args
                    putStrLn "== GHC Arguments: End =="